mappings/net/minecraft/server/command/ExecuteCommand.mapping

```diff
@@ -1,25 +1,51 @@
 CLASS sa net/minecraft/server/command/ExecuteCommand
-	CLASS sa$a Condition
+	CLASS sa$a ExistsCondition
+		METHOD test (Lcom/mojang/brigadier/context/CommandContext;)I
+			ARG 1 context
+	CLASS sa$b BooleanCondition
+		METHOD test (Lcom/mojang/brigadier/context/CommandContext;)Z
+			ARG 1 context
 	FIELD a BLOCKS_TOOBIG_EXCEPTION Lcom/mojang/brigadier/exceptions/Dynamic2CommandExceptionType;
 	FIELD b CONDITIONAL_FAIL_EXCEPTION Lcom/mojang/brigadier/exceptions/SimpleCommandExceptionType;
 	FIELD c CONDITIONAL_FAIL_COUNT_EXCEPTION Lcom/mojang/brigadier/exceptions/DynamicCommandExceptionType;
+	FIELD d BINARY_RESULT_CONSUMER Ljava/util/function/BinaryOperator;
 	METHOD a executeStoreScore (Lcd;Ljava/util/Collection;Lcsl;Z)Lcd;
 		ARG 0 source
 		ARG 1 targets
 		ARG 2 objective
+		ARG 3 requestResultValue
 	METHOD a executeStoreBossbar (Lcd;Lri;ZZ)Lcd;
+		ARG 0 source
 		ARG 1 bossBar
+		ARG 2 storeInValue
+		ARG 3 requestResultValue
 	METHOD a executeStoreData (Lcd;Lua;Lcr$h;Ljava/util/function/IntFunction;Z)Lcd;
+		ARG 0 source
+		ARG 1 object
+		ARG 2 path
+		ARG 3 tagSetter
+		ARG 4 requestResultValue
 	METHOD a register (Lcom/mojang/brigadier/CommandDispatcher;)V
-	METHOD a testObjectiveCondition (Lcom/mojang/brigadier/context/CommandContext;Lbi$d;)Z
-	METHOD a testObjectiveCondition (Lcom/mojang/brigadier/context/CommandContext;Ljava/util/function/BiPredicate;)Z
+		ARG 0 dispatcher
+	METHOD a testScoreMatch (Lcom/mojang/brigadier/context/CommandContext;Lbi$d;)Z
+		ARG 0 context
+		ARG 1 range
+	METHOD a testScoreCondition (Lcom/mojang/brigadier/context/CommandContext;Ljava/util/function/BiPredicate;)Z
+		ARG 0 context
+		ARG 1 condition
 	METHOD a executePositiveBlockCondition (Lcom/mojang/brigadier/context/CommandContext;Z)I
+		ARG 0 context
 		ARG 1 masked
 	METHOD a transformSourceForCondition (Lcom/mojang/brigadier/context/CommandContext;ZZ)Ljava/util/Collection;
+		ARG 0 context
 		ARG 1 positive
-		ARG 2 result
+		ARG 2 requestResultValue
 	METHOD a executeCondition (Lcom/mojang/brigadier/tree/CommandNode;Lcom/mojang/brigadier/builder/ArgumentBuilder;ZLsa$b;)Lcom/mojang/brigadier/builder/ArgumentBuilder;
-	METHOD a addBlockConditionExecutes (Lcom/mojang/brigadier/tree/CommandNode;Lcom/mojang/brigadier/builder/ArgumentBuilder;ZZ)Lcom/mojang/brigadier/builder/ArgumentBuilder;
+		ARG 0 node
+		ARG 1 builder
+	METHOD a addBlockConditionArguments (Lcom/mojang/brigadier/tree/CommandNode;Lcom/mojang/brigadier/builder/ArgumentBuilder;ZZ)Lcom/mojang/brigadier/builder/ArgumentBuilder;
```

Comment:
Why rename it? The method adds `executes`'s to the command, not arguments. It adds the execution actions for when there is no command argumebt and just a condition argument, to tell the player whether it matches

Reply:
Hmm, `executes` isn't a proper noun which is why the ethod name sounded very weird to me. Gonna change it to `Execute` tho, since it is only one that gets added. Also its used for `blocks`, not `block`. `block` and `scores` uses a method called `executeCondition` for the same thing.

Reply:
Two are being added. Also, you can name them "execute actions" or "subcommands" if you don't like "executes".

Reply:
No, it adds one, of a set of two, depending on `positive`.
```java
private static ArgumentBuilder<ServerCommandSource, ?> addBlocksConditionExecute(CommandNode<ServerCommandSource> node, ArgumentBuilder<ServerCommandSource, ?> builder, boolean positive, boolean masked) {
	return builder.fork(node, commandContext) -> transformSourceForCondition(commandContext, positive, testBlockCondition(commandContext, masked).isPresent())
	.executes(positive ?
		(commandContext -> executePositiveBlockCondition(commandContext, masked)) :
		(commandContext -> executeNegativeBlockCondition(commandContext, masked))
	);
}
```

Reply:
The fork is by the way, as far as I can tell, for the condition test, if there is a chained sub command. Forking a command makes it run for all sources. `transformSourceForCondition` actually returns an empty collection, or a collection with the source that was there before.
So the command gets forked with either an empty source or the source it already has, making sure the command will not continue running for any source at all, if the condition fails.
So the method actually takes care of both things. It stops the command if the condition fails, and it adds the execute for when there is no chained sub command.

Maybe `addBlocksConditionLogic` would be more fitting.